### DISCUSSION TITLE
Binary version of Spectre?

### DISCUSSION THREAD

**Discussion by hrj21** [ImmuneDynamics/Spectre#122] [Help]
Hi all! I hope you've had a good start to 2022! I'm running an R for cytometry course for people in my company, and we're using Spectre for one of the sessions. Our software is quite locked down by IT and my course attendees don't have Rtools on their work PCs and so I'm worried they won't be able to build Spectre. Is there an option to download and install a binary version of package version 1.0.0?

Many thanks
Hefin

**Comment by tomashhurst** [comment]
Hey @hrj21 , that's awesome! We don't have one up anywhere at the moment, but that should be pretty doable. @ghar1821 any chance you could take a quick look? Otherwise I'll try and sort it on Tuesday.

**Comment by hrj21** [comment]
Thanks Thomas. I might actually be being a bit silly here, does it work if I just zip up the Spectre directory from my R package library, and share that for people to install in RStudio from the .zip file? Is that a thing?

**Comment by hrj21** [comment]
I'm so sorry to spam you, but I got one of my attendees to try installing Spectre, and despite not having Rtools, the package was installed and works fine! I'm not sure I understand why, but this seems fine. Apologies if I've wasted any of your time. As always, thanks for an excellent package. On a slightly different note: do you have plans for adding Spectre to CRAN in the future?

**Reply by ghar1821** [reply]
We are still toying between bioconductor and CRAN at the moment. Initially we were gonna add it to both, but then it might be difficult to keep the versions consistent. So stay tuned!

**Comment by ghar1821** [comment]
Sorry folks, I was on leave and only got back online yesterday! 

I couldn't quite remember anymore why we need Rtools. It might have something to do with FItsne requiring it in the past. Regardless, I'm glad that it works out. I've built the binary version for mac and uploaded them to the release page (https://github.com/ImmuneDynamics/Spectre/releases/tag/v1.0.0) just in case you need them. I'll also do one for windows when I manage to get a hand on a windows machine.

**Comment by ghar1821** [comment]
@tomashhurst , I uploaded a binary file for Windows onto our release page on Github (https://github.com/ImmuneDynamics/Spectre/releases/tag/v1.0.0). If you have time, can you check that it works on the VRD? Devtools have this `check_win_release` function which sends the package to win-builder.r-project website and built it there for windows machine. It's very neat for those who don't have access to a working windows machine. The downside is that you still need a windows machine to properly test the binary release..
